### PR TITLE
dev-util/codeblocks: rm *.la files, rm masked USE=pch, add '-fno-strict-aliasing' to fix compile error (lto)

### PR DESCRIPTION
**Pull request by band-a-prend**
Closes: https://bugs.gentoo.org/845303
Closes: https://bugs.gentoo.org/858338

### CODE REVIEW

File: dev-util/codeblocks/codeblocks-20.03-r5.ebuild [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+		--disable-static \
+		$(use_with contrib boost-libdir "${ESYSROOT}/usr/$(get_libdir)") \
+		$(use_enable debug) \
+		$(use_enable pch) \
+		${CONF_WITH_LST}
+}
+
+src_install() {
+	default
+	find "${ED}" -name '*.la' -delete || die
```
Comment (Flowdalic) [inline dev-util/codeblocks/codeblocks-20.03-r5.ebuild:73]:
```suggestion
	find "${ED}" -type f -name '*.la' -delete || die
```
See https://devmanual.gentoo.org/eclass-reference/ltprune.eclass/index.html

File: dev-util/codeblocks/codeblocks-9999.ebuild [hunk 2]
```diff
@@ -56,6 +56,11 @@ src_configure() {
 		$(use_with contrib contrib-plugins all)
 }
 
+src_install() {
+	default
+	find "${ED}" -name '*.la' -delete || die
```
Comment (Flowdalic) [inline dev-util/codeblocks/codeblocks-9999.ebuild:61]:
```suggestion
	find "${ED}" -type f -name '*.la' -delete || die
```
See https://devmanual.gentoo.org/eclass-reference/ltprune.eclass/index.html

**Review by Flowdalic — CHANGES_REQUESTED** [review]

**Comment by Flowdalic** [comment]
Thanks for your contriubtion. I only have a minor comment :)

File: dev-util/codeblocks/codeblocks-20.03-r5.ebuild (hunk 1 above)
Comment (band-a-prend) [reply dev-util/codeblocks/codeblocks-20.03-r5.ebuild]: Updated.

Additionally drop masked USE=pch in both .ebuilds

File: dev-util/codeblocks/codeblocks-9999.ebuild (hunk 2 above)
Comment (band-a-prend) [reply dev-util/codeblocks/codeblocks-9999.ebuild]: Done.

**Comment by thesamesam** [comment]
looks good but wrt strict aliasing could you report those issues to squirrel like upstream mentioned on the bug? I saw https://github.com/albertodemichelis/squirrel/pull/257

**Comment by band-a-prend** [comment]
Sorry, this week I was out of my PC and can't to look up pre-history more detailed.
Could PR be merged into portage tree before reporting the issue to squirrel upstream?
But I think I will report it this week.

**Comment by thesamesam** [comment]
OK, can you sync live in a follow-up too? Thanks!

**Comment by band-a-prend** [comment]
> looks good but wrt strict aliasing could you report those issues to squirrel like upstream mentioned on the bug? 

Related upstream issue for `dev-lang/squirrel`: https://github.com/albertodemichelis/squirrel/issues/261

**Pull request closed without merging**